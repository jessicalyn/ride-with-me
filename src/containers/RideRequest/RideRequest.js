import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Mutation } from 'react-apollo';
import gql from 'graphql-tag';
import { Loader } from '../../components/Loader/Loader';

export class RideRequest extends Component {
  constructor() {
    super()
    this.state = {
      message: "I'd like to join your ride..."
    }
  }

  handleChange = (e) => {
    const { name, value } = e.target
    this.setState({ [name]: value })
  }

  render() {
    const { message } = this.state
    const { rideId } = this.props
    const { uuid } = this.props.user
    const CREATE_REQUEST = gql`
      mutation ($message: String!, $passengerUuid: String!, $rideId: Int!) {
        createRequest(message: $message, passengerUuid: $passengerUuid, rideId: $rideId) {
          request {
            id
            message
            passenger { id }
            ride { id }
          }
        }
      }`
    return(
      <div className="containers">
        <h2>Ride Request</h2>
        <form>
          <h3>Write a message to { this.props.driverName }:</h3>
          <textarea 
            name="message" 
            value={ this.state.message }
            onChange={ this.handleChange }
            rows="5"
            cols="33">
          </textarea>
          <Mutation mutation={CREATE_REQUEST} >
            {(createRequest, { data, loading, error }) => {
              if (loading) return <Loader />
              if (error) return <p>{ error }</p>
              if (!data) return <button className="standard-button request-button" onClick={(e) => {
                e.preventDefault()
                createRequest({ variables: {
                  "message": message,
                  "passengerUuid": uuid,
                  "rideId": rideId
                }})
              }}>Send Request</button>
              if (data) return <h3>Request Sent!</h3>
            }}
          </Mutation>
        </form>
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  user: state.user
})

export default connect(mapStateToProps)(RideRequest)