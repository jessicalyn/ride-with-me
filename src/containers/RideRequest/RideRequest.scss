textarea {
  padding: 10px;
  border-radius: 10px;
  width: 75%;
  height: 100px;
  font-family: $font-main;
  font-size: 16px;
  margin: 0;
  border: 1px solid #aaa;
  box-shadow: 0 1px 0 1px rgba(0,0,0,.04);
  border-radius: .5em;
  -moz-appearance: none;
  -webkit-appearance: none;
}

.request-button {
  width: 250px
}