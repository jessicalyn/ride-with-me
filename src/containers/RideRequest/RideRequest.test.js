import { RideRequest, mapStateToProps } from './RideRequest';
import React from 'react';
import { shallow } from 'enzyme';

describe('RideRequest Container', () => {
  describe('RideRequest', () => {
    let wrapper
    
    const mockState = {
      message: "I'd like to join your ride..."
    }
    const mockUser = {id: 1, name: "Jessica"}

    beforeEach(() => {
      wrapper = shallow(
        <RideRequest 
          user={mockUser}
        />
      )
    })

    it('should match the snapshot', () => {
      expect(wrapper.debug()).toMatchSnapshot()
    })

    it('should have proper default state', () => {
      expect(wrapper.state()).toEqual(mockState)
    })

    it('should set state on handleChange', () => {
      const mockEvent = { target: { name: "message", value : "Going to the strawbale festival!" }}
      const updatedState = {
        message: "Going to the strawbale festival!"
      }
      wrapper.instance().handleChange(mockEvent)

      expect(wrapper.state()).toEqual(updatedState)
    })

  })

  describe('mapStateToProps', () => {
    it('should return the expected props to state', () => {

      const mockState = {
        user: {id: 1, name: "Jessica"},
        fakeState: "Not real state to return"
      }
      const expected = {
        user: {id: 1, name: "Jessica"}
      }
  
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })
})