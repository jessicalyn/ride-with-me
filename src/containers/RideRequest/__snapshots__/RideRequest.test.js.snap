// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RideRequest Container RideRequest should match the snapshot 1`] = `
"<div className=\\"containers\\">
  <h2>
    Ride Request
  </h2>
  <form>
    <h3>
      Write a message to 
      :
    </h3>
    <textarea name=\\"message\\" value=\\"I\\\\'d like to join your ride...\\" onChange={[Function]} rows=\\"5\\" cols=\\"33\\" />
    <Mutation mutation={{...}}>
      [function]
    </Mutation>
  </form>
</div>"
`;
